test/expr/closure/trailing.swift

```diff
@@ -202,7 +203,82 @@ func r23036383(foo: Foo23036383?, obj: Foo23036383) {
+
+  if twoClosureArgs({}) {} {} // expected-error {{trailing closure requires parentheses for disambiguation in this context}} {{23-25=, }} {{27-27=)}}
+
+  if let _ = (foo?.map {$0+1}.map({$0+1}).map{$0+1}) {} // OK
+
+  if let _ = (foo?.map {$0+1}.map({$0+1})).map({$0+1}) {} // OK
+}
+
+func id<T>(fn: () -> T) -> T { return fn() }
+func any<T>(fn: () -> T) -> Any { return fn() }
+
+func testSR8736() {
+  if !id { true } { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if id { true } == true { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if true == id { true } { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if id { true } ? true : false { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if true ? id { true } : false { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if true ? true : id { false } { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if id { [false,true] }[0] { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if id { { true } } () { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if any { true } as! Bool { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if let _ = any { "test" } as? Int { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if any { "test" } is Int { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if let _ = id { [] as [Int]? }?.first { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if id { true as Bool? }! { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if case id { 1 } = 1 { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if case 1 = id { 1 } { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if case 1 = id { 1 } /*comment*/ { return } // expected-error {{trailing closure requires parentheses for disambiguation in this context}}
+  
+  if case (id { 1 }) = 1 { return } // OK
+  
+  if case 1 = (id { 1 }) { return } // OK
+  
+  if [id { true }].count == 0 { return } // OK
+  
+  if [id { true } : "test"].keys.count == 0 { return } // OK
+  
+  if "\(id { true })" == "foo" { return } // OK
+  
+  if (id { true }) { return } // OK
+  
+  if true { }
```

Comment:
Since `true` is a keyword, I'm not sure this is testing exactly what you want to test. Maybe do a `let myTrue = true` above?

Reply:
Oops, yeah that's a copy-paste error, 🤦‍♂. I updated the test